### ISSUE TITLE
Dead Code Injection + Debug Protection exposes internal variable names

### ISSUE THREAD

**Issue by AlexElting** [javascript-obfuscator/javascript-obfuscator#121]
If you obfuscate the following code:
```javascript
function test() {
    var importantVariableName = 'test';
    return importantVariableName.indexOf('test');
 }
```

with the following settings:
```javascript
{
    deadCodeInjection: true,
    deadCodeInjectionThreshold: 1,
    debugProtection: true,
}
```

the variable name `importantVariableName` can be seen in clear text in the obfuscated code. This is the case even if the `mangle` and `stringArrayEncoding` options are used.

The named variable appears to be part of dead code since it is not set anywhere, but the obfuscated function call on it is still `.indexOf('test')`.

Tested on version 0.11.2 and on https://javascriptobfuscator.herokuapp.com/

**Comment by sanex3339** [comment]
I'll release 0.12.2 version later today or tomorrow

**Comment by sanex3339** [comment]
Fixed in `javascript-obfuscator@0.12.2`
